pkg/helm/untar.go

```diff
@@ -30,6 +31,80 @@ func Untar(r io.Reader, dir string) error {
 	return untar(r, dir)
 }
 
+// Untar reads the compressed zip file from r and writes it into dir.
+func Unzip(r zip.Reader, dir string) error {
+	return unzip(r, dir)
+}
+
+func unzip(r zip.Reader, dir string) (err error) {
+	if err != nil {
+		return err
+	}
+	t0 := time.Now()
+	nFiles := 0
+	madeDir := map[string]bool{}
+	defer func() {
+		td := time.Since(t0)
+		if err == nil {
+			log.Printf("extracted zip into %s: %d files, %d dirs (%v)", dir, nFiles, len(madeDir), td)
+		} else {
+			log.Printf("error extracting zip into %s after %d files, %d dirs, %v: %v", dir, nFiles, len(madeDir), td, err)
+		}
+	}()
+
+	// Closure to address file descriptors issue with all the deferred .Close() methods
+	extractAndWriteFile := func(f *zip.File) error {
```

Comment:
Did you copy any of this code from stack overflow or another source like a blog post? If so please quote the URL here for crediting the author.